lib/simplecov/source_file.rb

```diff
@@ -221,9 +242,9 @@ def ensure_remove_undefs(file_lines)
     end
 
     def build_lines
-      coverage_exceeding_source_warn if coverage_data["lines"].size > src.size
+      coverage_exceeding_source_warn if lines_data.size > src.size
```

Comment:
In `0.21.2` I'm encountering an error on this line when executing via [rubycritic](https://rubygems.org/gems/rubycritic).  Turns out `coverage_data` is an Array ([as documented](https://github.com/simplecov-ruby/simplecov/pull/987/files#diff-81a03b411bc0f94f1d0d8c6ae5065779a9ec8aabb88c9a15f7f8c5f47811a3faR11)), and [was previously an Integer](https://github.com/simplecov-ruby/simplecov/commit/6e7a9ab3fa2c03ee86930596139270d822e2cc77), so did this line ever work?  Related to that, how does [`coverage_data.fetch(:branches, {})` (below)](https://github.com/simplecov-ruby/simplecov/pull/987/files#diff-81a03b411bc0f94f1d0d8c6ae5065779a9ec8aabb88c9a15f7f8c5f47811a3faR293) work?

Reply:
`coverage_data` is a Hash in this PR. So it's strange that you have problems. Maybe you have coverage data in some old format that is breaking the code?

Reply:
Update: I see that rubycritic is doing [a lot of stuff](https://github.com/whitesmith/rubycritic/blob/main/lib/rubycritic/analysers/coverage.rb) with simplecov internals. So I guess they will have to update the code if this PR is merged. They already do something similar [here](https://github.com/whitesmith/rubycritic/blob/main/lib/rubycritic/analysers/coverage.rb#L112) for example.

Reply:
@tycooon just to be clear, I'm not having any problem with this PR.  I think this PR will solve my problem 🙂 

Also for clarity: the error I encountered was `no implicit conversion of String into Integer (TypeError)`.  From my reading of the traceback it looks like this error is internal to `simplecov`.  Yes, it's possible that it's due to some sort of mishandling by `rubycritic` (which invokes `simplecov` through [`find_coverage_percentage`](https://github.com/whitesmith/rubycritic/blob/main/lib/rubycritic/analysers/coverage.rb#L33)).  But it seems smart to focus on `simplecov` first because the existing code doesn't seem to make sense (which is what I was describing above).  I'm looking forward to this PR helping to clarify `SourceFile`, and hopefully also fixing the bug.

<details>
<summary>Traceback</summary>
running simple_cov
Traceback (most recent call last):
	19: from {PATH}bin/rubycritic:23:in `<main>'
	18: from {PATH}bin/rubycritic:23:in `load'
	17: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/bin/rubycritic:10:in `<top (required)>'
	16: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/cli/application.rb:21:in `execute'
	15: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/commands/default.rb:19:in `execute'
	14: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/commands/default.rb:24:in `critique'
	13: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/analysers_runner.rb:29:in `run'
	12: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/analysers_runner.rb:29:in `each'
	11: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/analysers_runner.rb:32:in `block in run'
	10: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/analysers/coverage.rb:20:in `run'
	 9: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/core/analysed_modules_collection.rb:32:in `each'
	 8: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/core/analysed_modules_collection.rb:32:in `each'
	 7: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/analysers/coverage.rb:21:in `block in run'
	 6: from {PATH}lib/ruby/gems/2.7.0/gems/rubycritic-4.6.1/lib/rubycritic/analysers/coverage.rb:38:in `find_coverage_percentage'
	 5: from {PATH}lib/ruby/gems/2.7.0/gems/simplecov-0.21.2/lib/simplecov/source_file.rb:81:in `covered_percent'
	 4: from {PATH}lib/ruby/gems/2.7.0/gems/simplecov-0.21.2/lib/simplecov/source_file.rb:35:in `coverage_statistics'
	 3: from {PATH}lib/ruby/gems/2.7.0/gems/simplecov-0.21.2/lib/simplecov/source_file.rb:333:in `line_coverage_statistics'
	 2: from {PATH}lib/ruby/gems/2.7.0/gems/simplecov-0.21.2/lib/simplecov/source_file.rb:241:in `lines_strength'
	 1: from {PATH}lib/ruby/gems/2.7.0/gems/simplecov-0.21.2/lib/simplecov/source_file.rb:43:in `lines'
{PATH}lib/ruby/gems/2.7.0/gems/simplecov-0.21.2/lib/simplecov/source_file.rb:224:in `build_lines': no implicit conversion of String into Integer (TypeError)
</details>